optimum/intel/generation/modeling.py

```diff
@@ -403,3 +472,553 @@ def _convert_to_standard_cache(
             )
             for layer_past in past_key_value
         )
+
+
+class TSModelForSeq2SeqLM(OptimizedModel, GenerationMixin):
+    auto_model_class = AutoModelForSeq2SeqLM
+    export_feature = "text2text-generation"
+    main_input_name = "input_ids"
+    base_model_prefix = "torch_script_model"
+
+    def __init__(
+        self,
+        decoder,
+        encoder,
+        config: PretrainedConfig = None,
+        decoder_pkv: Optional[torch.jit.RecursiveScriptModule] = None,

```

Comment:
we shouldn't have to export the decoder twice but support both cases (with and without cache) with only one as done in https://github.com/huggingface/optimum-intel/blob/c48f06ad3303df8f1c530db6f953c23f6e71553c/optimum/intel/generation/modeling.py#L150